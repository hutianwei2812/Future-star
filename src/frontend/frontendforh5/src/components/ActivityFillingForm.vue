<template>
  <div>
    <title-bar :is-on-load="isOnLoad"></title-bar>
    <nav-bar :special-color="specialColor"></nav-bar>
    <el-page-header @back="goBack" content></el-page-header>
    <div>
      <h1>创建新的校友活动</h1>
      <el-card class="box-card" shadow="hover">
        <div slot="header" class="clearfix">
        </div>
        <div class="activityForm">
          活动名称：&nbsp;&nbsp;&nbsp;&nbsp;
          <el-input class="normalInputbox" v-model="activity_name" placeholder="请输入"></el-input>
          <br />
          <br />活动地点：&nbsp;&nbsp;&nbsp;&nbsp;
          <el-input class="normalInputbox" v-model="activity_location" placeholder="请输入"></el-input>
          <br />
          <br />活动描述：&nbsp;&nbsp;&nbsp;&nbsp;
          <el-input class="normalInputbox" v-model="activity_abstract" placeholder="请输入"></el-input>
          <br />
          <br />活动目标人数：&nbsp;&nbsp;&nbsp;&nbsp;
          <el-input-number size="medium" v-model="activity_peopleneed"></el-input-number>&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;
          活动时间：&nbsp;&nbsp;&nbsp;&nbsp;
          <el-date-picker
            type="datetimerange"
            v-model="activity_time"
            range-separator="至"
            start-placeholder="开始日期"
            end-placeholder="结束日期"
            align="right"
            value-format="yyyy-MM-dd HH:mm:ss"
            @change="format"
          ></el-date-picker>
          <br />
          <br />活动具体流程：&nbsp;&nbsp;&nbsp;&nbsp;
          <div class="nextLine"></div>
          <el-input
            class="textField"
            type="textarea"
            :autosize="{ minRows: 5}"
            placeholder="请输入内容"
            v-model="activity_items"
          ></el-input>
          <br />
          <br />活动注意事项：&nbsp;&nbsp;&nbsp;&nbsp;
          <div class="nextLine"></div>
          <el-input
            class="textField"
            type="textarea"
            :autosize="{ minRows: 5}"
            placeholder="请输入内容"
            v-model="activity_tips"
          ></el-input>
          <br />
          <br />
        </div>
      </el-card>
      <br />
      <br />
      <br />
      <br />
      <el-button
        @click="activity_create"
        type="primary"
        icon="el-icon-circle-plus-outline"
        class="operbutton"
      >确认并新建</el-button>
      <br />
      <br />
      <br />
      <br />
    </div>
  </div>
</template>

<script>
import TitleBar from '../components/TitleBar.vue'
import NavBar from '../components/NavBar.vue'
import axios from 'axios'
import Qs from 'qs'
export default {
  name: 'ActivityFillingForm',
  props: {},
  data() {
    return {
      isOnLoad: true,
      specialColor: 3,
      activity_name: '',
      activity_time: '',
      activity_location: '',
      activity_abstract: '',
      activity_peopleneed: 0,
      activity_status: '',
      activity_tips: '',
      activity_items: '',
      targetURL: 'http://117.50.34.200:8002/createActivity/'
    }
  },
  methods: {
    format(val) {
      this.activity_time = val
    },
    test() {
      this.goBack()
    },
    goBack() {
      this.$router.push({
        name: 'ActivityManagePage'
      })
    },
    activity_create() {
      const _this = this
      if (
        _this.activity_name !== '' &&
        _this.activity_time !== '' &&
        _this.activity_location !== '' &&
        _this.activity_abstract !== '' &&
        _this.activity_peopleneed !== 0 &&
        _this.activity_tips !== '' &&
        _this.activity_items !== ''
      ) {
        axios({
          url: _this.targetURL,
          method: 'post',
          data: {
            activity_name: _this.activity_name,
            activity_time: _this.activity_time.join(','),
            activity_location: _this.activity_location,
            activity_abstract: _this.activity_abstract,
            activity_peopleneed: _this.activity_peopleneed,
            activity_tips: _this.activity_tips,
            activity_status: 'N',
            activity_items: _this.activity_items
          },
          transformRequest: [
            function(data) {
              return Qs.stringify(data)
            }
          ]
        }).then(response => {
          if (response.data.code === '1') {
            this.$alert('活动创建成功！', '页面提示您：')
            this.$emit('turnPage')
            this.test()
          }
        })
      } else {
        this.$alert('表单信息填写不全，请填写完毕后再提交！', '页面提示您：')
      }
    }
  },
  created() {
    let adminName = localStorage['name']
    let adminPwd = localStorage['pwd']
    if (typeof adminName !== 'undefined' && adminName !== '') {
      axios({
        url: 'http://117.50.34.200:8002/adminForLogin/',
        method: 'post',
        data: {
          admin_name: adminName,
          admin_password: adminPwd
        },
        transformRequest: [
          function(data) {
            return Qs.stringify(data)
          }
        ]
      }).then(response => {
        if (response.data.code === 'ok') {
          let power = localStorage['powerActivity']
          if (power === 'true') {
          } else if (power === 'false') {
            this.$router.push({
              name: 'ActivityManagePage'
            })
            this.$message({
              type: 'warning',
              message: '权限不足!'
            })
          }
        } else if (response.data.code === 'no') {
          this.$store.commit('setName', '')
          this.$store.commit('setPwd', '')
          this.$store.commit('setPowerCamp', 'false')
          this.$store.commit('setPowerCourse', 'false')
          this.$store.commit('setPowerActivity', 'false')
          this.$store.commit('setPowerAdministrator', 'false')
          this.$router.push({
            name: 'Index'
          })
        } else {
          this.$notify,
            error({
              title: 'Error',
              message: response.data
            })
          this.$store.commit('setName', '')
          this.$store.commit('setPwd', '')
          this.$store.commit('setPowerCamp', 'false')
          this.$store.commit('setPowerCourse', 'false')
          this.$store.commit('setPowerActivity', 'false')
          this.$store.commit('setPowerAdministrator', 'false')
          this.$router.push({
            name: 'Index'
          })
        }
      })
    } else {
      this.$router.push({
        name: 'Index'
      })
    }
  },
  components: {
    TitleBar,
    NavBar
  }
}
</script>


<style scoped>
.textField {
  width: 85%;
}
.normalInputbox {
  width: 75%;
}
.text {
  font-size: 20px;
}
.item {
  margin-bottom: 18px;
  height: 30px;
}
.clearfix:before,
.clearfix:after {
  display: table;
  content: '';
}
.clearfix:after {
  clear: both;
}
.box-card {
  top: 20px;
  width: 1100px;
  position: relative;
  margin: auto;
}
.operbutton {
  width: 160px;
  left: 45%;
  position: relative;
  margin: auto;
}
.activityForm {
  text-align: center;
}
.nextLine {
  margin: 10px 0;
}
h1 {
  text-align: center;
}
</style>