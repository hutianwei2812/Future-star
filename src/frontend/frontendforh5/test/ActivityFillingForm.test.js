import { mount } from '@vue/test-utils'
import ActivityFillingForm from '@/components/ActivityFillingForm'
import Vue from 'vue'
import ElementUI from 'element-ui'
Vue.use(ElementUI)

test('renders correctly', () => {
  const wrapper = mount(ActivityFillingForm)
  expect(wrapper.element).toMatchSnapshot()
})