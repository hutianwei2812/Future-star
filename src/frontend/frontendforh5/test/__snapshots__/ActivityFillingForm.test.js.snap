// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <header
    class="el-header"
    style="height: 60px;"
  >
    <span
      class="title-span"
    >
      未来之星管理系统
    </span>
     
    <span
      class="user-span"
    >
      
    </span>
     
    <a
      class="el-link el-link--primary is-underline"
    >
      <!---->
      <span
        class="el-link--inner"
      >
        注销
      </span>
      <!---->
    </a>
  </header>
   
  <div
    class="nav"
  >
    <section
      class="el-container is-vertical"
    >
      <header
        class="el-header"
        style="height: 60px;"
      >
        <div
          class="newDiv"
        >
          <ul
            class="sub_nav"
          >
            <div>
              <li
                class="sub_nav_li"
                id="normalColor"
              >
                报名管理
              </li>
            </div>
             
            <div>
              <li
                class="sub_nav_li"
                id="normalColor"
              >
                课程管理
              </li>
            </div>
             
            <div>
              <li
                class="sub_nav_li"
                id="blueColor"
              >
                活动管理
              </li>
            </div>
             
            <div>
              <li
                class="sub_nav_li"
                id="normalColor"
              >
                校友管理
              </li>
            </div>
             
            <div>
              <li
                class="sub_nav_li"
                id="normalColor"
              >
                管理员权限设置
              </li>
            </div>
          </ul>
        </div>
      </header>
    </section>
  </div>
   
  <div
    class="el-page-header"
  >
    <div
      class="el-page-header__left"
    >
      <i
        class="el-icon-back"
      />
      <div
        class="el-page-header__title"
      >
        返回
      </div>
    </div>
    <div
      class="el-page-header__content"
    >
      新建活动
    </div>
  </div>
   
  <div>
    <div
      class="el-card box-card is-hover-shadow"
    >
      <div
        class="el-card__header"
      >
        <div
          class="clearfix"
        >
          <span>
            创建新的校友活动
          </span>
        </div>
      </div>
      <div
        class="el-card__body"
      >
         
        <div
          class="activityForm"
        >
          
        活动名称：    
          <div
            class="normalInputbox el-input"
          >
            <!---->
            <input
              autocomplete="off"
              class="el-input__inner"
              placeholder="请输入"
              type="text"
            />
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <br />
          <br />
          
        活动地点：    
          <div
            class="normalInputbox el-input"
          >
            <!---->
            <input
              autocomplete="off"
              class="el-input__inner"
              placeholder="请输入"
              type="text"
            />
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <br />
          <br />
          
        活动描述：    
          <div
            class="normalInputbox el-input"
          >
            <!---->
            <input
              autocomplete="off"
              class="el-input__inner"
              placeholder="请输入"
              type="text"
            />
            <!---->
            <!---->
            <!---->
            <!---->
          </div>
          <br />
          <br />
          
        活动目标人数：    
          <div
            class="el-input-number el-input-number--medium"
          >
            <span
              class="el-input-number__decrease"
              role="button"
            >
              <i
                class="el-icon-minus"
              />
            </span>
            <span
              class="el-input-number__increase"
              role="button"
            >
              <i
                class="el-icon-plus"
              />
            </span>
            <div
              class="el-input el-input--medium"
            >
              <!---->
              <input
                aria-disabled="undefined"
                aria-valuemax="Infinity"
                aria-valuemin="-Infinity"
                aria-valuenow="0"
                autocomplete="off"
                class="el-input__inner"
                max="Infinity"
                min="-Infinity"
                role="spinbutton"
                type="text"
              />
              <!---->
              <!---->
              <!---->
              <!---->
            </div>
          </div>
          
                   
        活动时间：    
          <div
            class="el-date-editor el-range-editor el-input__inner el-date-editor--datetimerange"
          >
            <i
              class="el-input__icon el-range__icon el-icon-time"
            />
            <input
              autocomplete="off"
              class="el-range-input"
              name=""
              placeholder="开始日期"
            />
            <span
              class="el-range-separator"
            >
              至
            </span>
            <input
              autocomplete="off"
              class="el-range-input"
              name=""
              placeholder="结束日期"
            />
            <i
              class="el-input__icon el-range__close-icon"
            />
          </div>
          <br />
          <br />
          
        活动具体流程：    
        
          <div
            class="nextLine"
          />
           
          <div
            class="textField el-textarea"
          >
            <textarea
              autocomplete="off"
              class="el-textarea__inner"
              placeholder="请输入内容"
            />
            <!---->
          </div>
          <br />
          <br />
          
        活动注意事项：    
        
          <div
            class="nextLine"
          />
           
          <div
            class="textField el-textarea"
          >
            <textarea
              autocomplete="off"
              class="el-textarea__inner"
              placeholder="请输入内容"
            />
            <!---->
          </div>
           
          <br />
          <br />
        </div>
      </div>
    </div>
     
    <br />
    <br />
    <br />
    <br />
     
    <button
      class="el-button operbutton el-button--primary"
      type="button"
    >
      <!---->
      <i
        class="el-icon-circle-plus-outline"
      />
      <span>
        确认并新建
      </span>
    </button>
     
    <br />
    <br />
    <br />
    <br />
  </div>
</div>
`;
